mne/io/base.py

```diff
@@ -1086,17 +1087,47 @@ def apply_function(
         if dtype is not None and dtype != self._data.dtype:
             self._data = self._data.astype(dtype)
 
+        args = getfullargspec(fun)[0]
```

Comment:
This isn't quite enough... you need to check `.kwonlyargs` as well

Reply:
done